src/glum/_glm.py

```diff
@@ -831,6 +884,17 @@ def _tear_down_from_fit(self):
         del self._solver
         del self._random_state
 
+    def _set_up_for_info_criteria(self, **kwargs):
```

Comment:
I'm not sure I understand the need for this method. It's only called once in the `fit` method and will always have the same argument (`y`).